src/Common/Config/YAMLParser.cpp

```diff
@@ -0,0 +1,196 @@
+#include "YAMLParser.h"
+
+#include <string>
+#include <cstring>
+#include <vector>
+
+#include <Poco/DOM/Document.h>
+#include <Poco/DOM/DOMParser.h>
+#include <Poco/DOM/DOMWriter.h>
+#include <Poco/DOM/NodeList.h>
+#include <Poco/DOM/Element.h>
+#include <Poco/DOM/AutoPtr.h>
+#include <Poco/DOM/NamedNodeMap.h>
+#include <Poco/DOM/Text.h>
+#include <Common/Exception.h>
+
+#include <yaml-cpp/yaml.h>
+
+#include <common/logger_useful.h>
+
+using namespace Poco::XML;
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int CANNOT_PARSE_TEXT;
+    extern const int UNKNOWN_EXCEPTION;
+}
+
+// A prefix symbol in yaml key
+const char YAML_ATTRIBUTE_PREFIX = '@';
+
+YAMLParser::YAMLParser() = default;
+
+Poco::AutoPtr<Poco::XML::Document> YAMLParser::parse(const String& path)
+{
+    YAML::Node node_yml;
+    try
+    {
+        node_yml = YAML::LoadFile(path);
+    }
+    catch (const YAML::ParserException&)
+    {
+        /// yaml-cpp cannot parse the file because its contents are incorrect
+        throw Exception(ErrorCodes::CANNOT_PARSE_TEXT, "Unable to parse YAML configuration file {}", path);
```

Comment:
YAML exceptions are quite informative (they contain a row and a column and the exact reason of fail too), so it's better to keep their messages but wrap it in our exceptions, like this

```
catch (const YAML::BadFile & e)
{
  throw Exception(ErrorCodes::CANNOT_OPEN_FILE, "Couldn't open YAML configuration file {}", path);
}
catch (const YAML::Exception & e)
{
  throw Exception(ErrorCodes::CANNOT_PARSE_YAML, "Couldn't parse YAML configuration file {}: {}", path, e.what());
}
```